File: quizvocabulary-server/src/main/java/com/quizvocabulary/data/model/Study.java
package com.quizvocabulary.data.model;

import java.util.List;
import java.util.UUID;

import lombok.Data;

@Data
public class Study {

	private UUID id;
	
	private Language languageA;
	
	private Language languageB;

	private String profileId;

	private Board defaultBoard;

	private List<Board> boards;
}

File: quizvocabulary-ui/src/main/quizvocabulary-ui/src/app/model/study.ts
import { Board } from './board';
import { Language } from './language';

export class Study {

  id: number;
  language: Language;
  profileLanguage: Language;
  defaultBoard: Board;

  constructor(language: Language, profileLanguage: Language) {
    this.language = language;
    this.profileLanguage = profileLanguage;
    this.defaultBoard = new Board('Main Board', language, profileLanguage);
  }
}
